src/linear_algebra/general_linear_group.lean

```diff
@@ -0,0 +1,263 @@
+/-
+
+The Genera Linear group $GL(n, R)$
+-/
+import linear_algebra.matrix
+import linear_algebra.matrix.nonsingular_inverse
+import linear_algebra.special_linear_group
+import linear_algebra.determinant
+
+
+/-!
+# The General Linear group $GL(n, R)$
+
+This file defines the elements of the General Linear group `general_linear_group n R`,
+consisting of all invertible `n` by `n` `R`-matrices.
+
+
+## Main definitions
+
+ * `matrix.GL` is the type of matrices over R with unit determinant
+ * `matrix.GL.group` gives the group structure (under multiplication)
+ * `matrix.GL_plus.GL_pos` gives the subgroup of matrices with
+ positive determinant (over a linear ordered ring)
+
+## Implementation notes
+
+
+## References
+
+
+## Tags
+
+matrix group, group, matrix inverse
+-/
+
+namespace matrix
+universes u v
+open_locale matrix
+open linear_map
+
+
+section
+
+variables (n : Type u) [decidable_eq n] [fintype n] (R : Type v) [comm_ring R]
+
+/-- `GL n R` is the group of `n` by `n` `R`-matrices with unit determinant.
+Defined as a subtype of matrices
+-/
+abbreviation GL (R : Type*) [comm_ring R] :  Type* := units (matrix n n R )
+
+
+
+
+
+lemma GL.is_unit_det  (A : GL n R) : is_unit ( (A : matrix n n R).det) :=
+begin
+  rw ←matrix.is_unit_iff_is_unit_det,
+  exact units.is_unit _
+end
+
+
+/--The `GL n R` and `general_linear_group R n` are multiplicatively equivalent-/
+def GL_mul_equiv_general_linear_group : (GL n R) ≃* (linear_map.general_linear_group R ( n → R)) :=
+units.map_equiv to_lin_alg_equiv'.to_mul_equiv
+
+/--Given a matrix with unit determinant we get an element of `GL n R`-/
+noncomputable def GL.mk' (A: matrix n n R) (h: is_unit (det A)): GL n R:=
+⟨A, nonsing_inv A, by {apply mul_nonsing_inv, apply h,}, by  {apply nonsing_inv_mul, apply h, }⟩
```

Comment:
If you use `matrix.unit_of_det_invertible` here and replace `is_unit` with `invertible` then this becomes computable.

Reply:
I added a computable version of this using your suggestion. I did want to leave this one also, as for some of the other things I have I found it useful.